index.js

```diff
@@ -110,8 +110,9 @@ Prober.prototype.probe = function probe(request, bypass, callback) {
     var wrappedCallback;
     if (this.detectFailuresByCallback) {
         wrappedCallback = function(err, resp) {
-            var errResponse = resp && !isNaN(resp.statusCode) &&
-                resp.statusCode >= 500;
+            var errResponse = resp &&
+                (!isNaN(resp.statusCode) && resp.statusCode >= 500 || // HTTP
+                resp.hasOwnProperty('ok') && !resp.ok); // tchannel
```

Comment:
@Raynos @syyang instead of creating a `tchannelProbe` function, I made the health check function customizable, so that airlock can support other protocols easily, without having to add a specific function for tchannel. please take a another look.